File: networkproj/src/main/java/com/example/telecominventory/service/AssetService.java
package com.example.telecominventory.service;

import com.example.telecominventory.entity.Asset;
import com.example.telecominventory.entity.AssetStatus;
import com.example.telecominventory.entity.AssetType;

import java.util.List;
import java.util.Optional;

public interface AssetService {

    // ==========================
    // 🔹 BASIC CRUD OPERATIONS
    // ==========================
    List<Asset> getAllAssets();
    Optional<Asset> getAssetById(Long id);
    Asset createAsset(Asset asset);
    Asset updateAsset(Long id, Asset asset);
    void deleteAsset(Long id);

    // ==========================
    // 🔹 FETCH / FILTER OPERATIONS
    // ==========================
    List<Asset> getAssetsByType(AssetType type);
    List<Asset> getAssetsByStatus(AssetStatus status);
    List<Asset> getAssetsByLocation(String location);

    // ==========================
    // 🔹 ASSIGNMENT OPERATION
    // ==========================
    Asset assignAsset(Long id, Long customerId, String performedBy);
}

File: networkreact/networkinventory/src/features/assets/services/assetService.js
import api from "../../../api/axiosConfig";

export const getAllAssets = () => api.get("/assets");

export const getAssetById = (id) => api.get(`/assets/${id}`);

export const createAsset = (asset) => api.post("/assets", asset);

export const updateAsset = (id, asset) => api.put(`/assets/${id}`, asset);

export const deleteAsset = (id) => api.delete(`/assets/${id}`);
